File: pyjl/tests/tests_py2many-py2many/classes.jl
struct Foo
end
function bar(self::Foo)::Int64
return baz(self)
end

function baz(self::Foo)::Int64
return 10
end

struct Person
name::String
end
function __init__(self::Person, name::String)
self.name = name
end

function get_name(self::Person)
return self.name
end

struct Student
student_number::Int64
end
function __init__(self::Student, name::String, student_number::Int64)
__init__(super(), name);
self.student_number = student_number
end

function get_name(self::Student)
return join([string(self.student_number), " - ", string(self.name)], "")
end

function main()
f = Foo()
b = bar(f)
@assert(b == 10)
p = Person("Michael Carmichael")
s = Student("Michael Carmichael", 111111)
@assert(get_name(p) == "Michael Carmichael")
@assert(get_name(s) == "111111 - Michael Carmichael")
println("OK");
end

main()

File: pyjl/tests/tests_py2many/classes.py
class Foo:
    def bar(self):
        return self.baz()

    def baz(self) -> int:
        return 10

# class Bag:
#     def __init__(self):
#         self.data = []

#     def add(self, x):
#         self.data.append(x)

#     def addtwice(self, x):
#         self.add(x)
#         self.add(x)

class Person:
    def __init__(self, name:str):
        self.name = name

    def get_name(self):
        return self.name

class Student(Person):
    def __init__(self, name:str, student_number:int):
        super().__init__(name)
        self.student_number = student_number

    def get_name(self):
        return f"{self.student_number} - {self.name}"

if __name__ == "__main__":
    f = Foo()
    b = f.bar()
    assert b == 10

    #  Test inheritance
    p = Person("Michael Carmichael")
    s = Student("Michael Carmichael", 111111)
    assert p.get_name() == "Michael Carmichael"
    assert s.get_name() == "111111 - Michael Carmichael"

    print("OK")
